### PR TITLE
Replace urlopen with requests in openlibrary/views/showmarc

### PR DESCRIPTION
**Pull request by dherbst**
Related #2852 

Refactor showmarc to replace urlopen with requests.

### Technical

Refactor showmarc to replace urlopen with requests.
Some pylint and pycodestyle changes for formatting.

### Testing

```python
r0 = 0
r1 = 894
filename = 'CollingswoodLibraryMarcDump10-27-2008/Collingswood.out'
url = 'http://www.archive.org/download/%s' % filename
headers = {'Range': 'bytes=%d-%d' % (r0, r1)}
response = requests.get(url, headers=headers)
response.raise_for_status()
```

### Screenshot

N/A

### Stakeholders

@cclauss @hornc

### CODE REVIEW

File: openlibrary/views/showmarc.py [hunk 1]
```diff
@@ -106,19 +112,17 @@ def GET(self, filename, offset, length):
         offset = int(offset)
         length = int(length)
 
-        #print "record_locator: <code>%s</code><p/><hr>" % locator
+        # print "record_locator: <code>%s</code><p/><hr>" % locator
```
Comment (cclauss) [inline openlibrary/views/showmarc.py:115]:
```suggestion
        # print("record_locator: <code>%s</code><p/><hr>" % locator)
```

File: openlibrary/views/showmarc.py [hunk 2]
```diff
@@ -106,19 +112,17 @@ def GET(self, filename, offset, length):
...
-                               )
+        url = 'http://www.archive.org/download/%s' % filename
+        headers = {'Range': 'bytes=%d-%d' % (r0, r1)}
 
         try:
-            result = urllib.request.urlopen(ureq).read(100000)
-        except urllib.error.HTTPError as e:
+            response = requests.get(url, headers=headers)
+            response.raise_for_status()
+            result = response.content
```
Comment (cclauss) [inline openlibrary/views/showmarc.py:124]:
```suggestion
            result = response.content[:100000]
```

**Review by cclauss — APPROVED** [review]

File: openlibrary/views/showmarc.py [hunk 3]
```diff
@@ -106,19 +112,17 @@ def GET(self, filename, offset, length):
...
+        # print("record_locator: <code>%s</code><p/><hr>" % locator)
 
         r0, r1 = offset, offset+100000
-        url = 'http://www.archive.org/download/%s'% filename
-
-        ureq = urllib.request.Request(url,
-                               None,
-                               {'Range':'bytes=%d-%d'% (r0, r1)},
-                               )
+        url = 'http://www.archive.org/download/%s' % filename
```
Comment (hornc) [inline openlibrary/views/showmarc.py:118]: @dherbst looks good! I just noticed the URLs in this file are old and take two redirects to get to the current:
`https://archive.org/...` URL

File: openlibrary/views/showmarc.py [hunk 4]
```diff
@@ -106,19 +112,17 @@ def GET(self, filename, offset, length):
         offset = int(offset)
         length = int(length)
 
-        #print "record_locator: <code>%s</code><p/><hr>" % locator
+        # print("record_locator: <code>%s</code><p/><hr>" % locator)
```
Comment (hornc) [inline openlibrary/views/showmarc.py:115]: This comment out print can be removed.

File: openlibrary/views/showmarc.py (hunk 3 above)
Comment (cdrini) [reply openlibrary/views/showmarc.py:118]:
```suggestion
        url = 'https://www.archive.org/download/%s' % filename
```

Comment (hornc) [reply openlibrary/views/showmarc.py:118]: @cdrini It needs to be  `url = 'https://archive.org/download/%s' % filename`  , `www` is a redirect also.

**Review by cclauss — APPROVED** [review]

**Review by mekarpeles — COMMENTED** [review]
lgtm, over to @cclauss test + merge when you're able :)

**Comment by cclauss** [comment]
@hornc Do you approve?

**Comment by hornc** [comment]
@cclauss The URLs still result in unnecessary redirects -- all archive.org URLs should be  `https:` and without the `www`  as per my earlier comment, but this will work with the requests change.

**Comment by dherbst** [comment]
I must have overlooked the URL comment, sorry.  I'll update it later tonight.

**Review by hornc — APPROVED** [review]
Thanks @dherbst ! @cclauss looks good to merge!

**Pull request merged**